From 5760ce93e55ad139dc562f5169caf4a442703413 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 26 Dec 2022 09:13:29 +0200
Subject: [PATCH] [SW-113135] gaudi3: change dccm base to internal address

Currently, the DCCM base used by ppboot for ARC1 is using
DMI not internal. This will cause ARC stall issue when was also
seen in gaudi2 when ppboot will access arcboot rom api's and
ppboot stack is using DMI address vs rom api using internal
address for DMI. To avoid ARC stall issue this patch changes
DCCM base to internal address with change in offset to match
latest rom DCCM usage.

Change-Id: I9af757f8155381657879b49805e7179419e8d205
---

diff --git a/arch/gaudi3/arc/common.h b/arch/gaudi3/arc/common.h
index d5d705e..0e0cc7d 100644
--- a/arch/gaudi3/arc/common.h
+++ b/arch/gaudi3/arc/common.h
@@ -57,10 +57,10 @@
 #define BOOT_INFO_BASE			SP_SRAM_BASE
 
 /* In order not to conflict with ARC BOOTROM's usage of DCCM,
- * Let PPBoot start from the offset of 0x5000, for using DCCM
+ * Let PPBoot start from the offset of 0x6300, for using DCCM
  * for its own BSS and Stack.
  */
-#define PPBOOT_DCCM_BASE		(ARC1_DCCM_BASE + 0x5000)
+#define PPBOOT_DCCM_BASE		0x80006300
 
 #define FUSE_WORDS_PER_BANK		0x80
 #define I2C_SLAVE_ADDRESS		0x6A
